### DISCUSSION TITLE
Add comment syntax

### DISCUSSION THREAD

**Discussion by idrm** [tailwindlabs/tailwindcss#9085] [Ideas]
I have cases where a single element's style definition is made up of dozens of lines of tailwind directives (e.g. a button component that has specific styling based on `data-` attributes, its disabled state, etc.). An ability to add comments at different places inside my definitions would make it easier for others and myself, at a later time, to understand when or why particular rules apply. 

C-style comment syntax, as in the following example, could be one option: 

```
/* default state */
text-red-500 bg-gray-200

/* disabled state */
disabled:(text-red-100 bg-gray-700)
```

I suppose this addition won't be as handy when used inside plain HTML files, but in my case, where I use TailwindCSS together with a CSS-in-JS library inside a web app, multiline style definitions reduce clutter since they appear separately from where the styled components are used. E.g. my code looks something like this (the example I give is greatly shortened compared to my actual `Button` definition):

```jsx
const Button = styled.button([
  tw`
    text-blue-500 bg-gray-200
    hover:(text-blue-700 bg-gray-300)    
    disabled:(
      text-blue-100 bg-gray-50
      hover:(text-blue-100 bg-gray-50)
    )
    [&[data-color=danger]]:(
      text-red-500
      hover:(text-red-700)
      disabled:(
        text-red-100
        hover:(text-red-100)
      )
    )
  `
])
```

**Comment by wongjn** [comment]
With regards to the grouping, [here's a response to a similar suggestion](https://github.com/tailwindlabs/tailwindcss/discussions/9053#discussioncomment-3350470):

> We get this question so many times, so I would like to redirect you here for more information: https://twitter.com/adamwathan/status/1461519820411789314
